app/uk/gov/hmrc/taxenrolmentassignmentfrontend/connectors/IVConnector.scala

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2022 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.taxenrolmentassignmentfrontend.connectors
+
+import cats.data.EitherT
+import com.google.inject.{Inject, Singleton}
+import play.api.http.Status.OK
+import uk.gov.hmrc.http.{HeaderCarrier, HttpClient, HttpResponse}
+import uk.gov.hmrc.service.TEAFResult
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.config.AppConfig
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.errors.UnexpectedResponseFromIV
+import uk.gov.hmrc.taxenrolmentassignmentfrontend.models.IVNinoStoreEntry
+import uk.gov.hmrc.http.HttpReads.Implicits._
+
+import scala.concurrent.ExecutionContext
+
+@Singleton
+class IVConnector @Inject()(httpClient: HttpClient, appConfig: AppConfig) {
+
+  def getCredentialsWithNino(nino: String)(
+    implicit ec: ExecutionContext,
+    hc: HeaderCarrier
+  ): TEAFResult[List[IVNinoStoreEntry]] = EitherT {
+    val url = s"${appConfig.IV_BASE_URL}/nino"
+    httpClient
+      .GET[HttpResponse](url, queryParams = Seq("nino" -> nino))
+      .map(
+        httpResponse =>
+          httpResponse.status match {
+            case OK => Right(httpResponse.json.as[List[IVNinoStoreEntry]])
+            case _  => Left(UnexpectedResponseFromIV)
```

Comment:
Will this work? Just thinking about http verbs and it's conversion of some status's into the UpstreamErrorResponse Exception. So matching on HttpResponse status's might lead to this being unreachable? (I haven't verified whether this thought is applicable here yet)

Reply:
![image](https://user-images.githubusercontent.com/29196699/148918827-9afd1923-98a9-4203-8179-ae91b64b2c27.png)

Reply:
The tests showed this work. I think because it is httpResponse rather than the model itself. The function could be changed to:

`httpClient
      .GET[List[IVNinoStoreEntry]](url, queryParams = Seq("nino" -> nino))
      .map(Right(_))
      .recover {
        case ex: HttpException =>
          Left(UnexpectedResponseFromIV)
      }`
This change also required no change with the tests so parsing the response using httpVerbs creates the exception.

Reply:
I made this comment before I got to the Integration Tests, if they pass those then it's shouldn't be an issue. I'm not familiar with Cats so I'll have to defer to your experience.